lib/sync/sync-processor.js

```diff
@@ -95,15 +95,18 @@ function syncWithBackend(payload, tries, callback) {
   async.waterfall([
     function setSyncStart(cb) {
       var syncLoopStartTime = Date.now();
-      syncStorage.updateDatasetClient(datasetClientId, {syncLoopStart: syncLoopStartTime, syncLoopEnd: syncLoopStartTime + expectedTimeout}, function(err){
-        return cb(err);
+      syncStorage.updateDatasetClient(datasetClientId, {syncLoopStart: syncLoopStartTime, syncLoopEnd: syncLoopStartTime + expectedTimeout}, function(err, datasetClient){
+        return cb(err, datasetClient);
       });
     },
-    function listData(cb) {
-      listDataWithTimeout({datasetClientId: datasetClientId, datasetId: datasetId, queryParams: queryParams, metaData: metaData}, expectedTimeout, cb);
+    function listData(datasetClient, cb) {
+      listDataWithTimeout({datasetClientId: datasetClientId, datasetId: datasetId, queryParams: queryParams, metaData: metaData}, expectedTimeout, function(err, res) {
+        return cb(err, datasetClient, res);
+      });
     },
-    function saveRecords(recordsWithHash, cb) {
-      var globalHash = hashProvider.globalHash(datasetId, _.pluck(recordsWithHash, 'hash'));
+    function saveRecords(datasetClient, recordsWithHash, cb) {
+      var toHash = _.pluck(recordsWithHash, 'hash').concat([datasetClient.collisionCount]);
+      var globalHash = hashProvider.globalHash(datasetId, toHash);
```

Comment:
I am wondering should we just suffix the collision count value to the global hash, rather than add it to the records hashes and then hash it, so the new global hash will be like:

`23423498hsdkkf_1`

The reason is that it will allow us to see which is causing the global hash to change - is it because of records changes, or because of collision changes. This way will make it easy to find out.

WDYT?